bin/pyenv-installer

```diff
@@ -37,18 +37,41 @@ if ! command -v git 1>/dev/null 2>&1; then
   exit 1
 fi
 
-if [ -n "${USE_GIT_URI}" ]; then
-  GITHUB="git://github.com"
+# Check ssh authentication if USE_SSH_URL is present
+if [ -n "${USE_SSH_URL}" ]; then
+  if ! command -v ssh 1>/dev/null 2>&1; then
+    echo "pyenv: configuration USE_SSH found but ssh is not installed, can't continue." >&2
+    exit 1
+  fi
+
+  ssh -T git@github.com 1>/dev/null 2>&1 || EXIT_CODE=$?
+  if [[ ${EXIT_CODE} != 1 ]]; then
```

Comment:
Need a comment here with the obscure piece of knowledge that this uses. Smth like:

> `ssh -T git@github.com' returns 255 on authentication failure and 1 on success. See https://docs.github.com/en/repositories/creating-and-managing-repositories/troubleshooting-cloning-errors#check-your-ssh-access

That said, that doc doesn't specify the exit codes! Would if be more reliable to check for the "successfully authenicated" message?

Reply:
I just found out that Github mention the exit code in another doc.

> "Note: The remote command should exit with code 1."
> -- [Github Docs - testing-your-ssh-connection](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection)

Reply:
99f5a9469c5f4e01adcf4175fc4efe4e419a967f